File: src/web/src/api/coveredCall.ts
module Saturn.Api {
    export class CoveredCallType {
        public static ITM = "ITM"; // in the money
        public static OTM = "OTM"; // out of the money
    }

    export interface CoveredCall {
        option: Option;
        type: CoveredCallType;
    }

    export interface ITMCoveredCall extends CoveredCall {
        breakEven: number;
        downsideProtection: number;
        roi: number;
    }

    export interface OTMCoveredCall extends CoveredCall {
        expectedRoi: number;
        upsideRequired: number;
    }
}

File: src/main/com/saturn/api/strategy/CoveredCall.java
package com.saturn.api.strategy;

import com.saturn.api.core.Option;

public class CoveredCall {
	private final Option option;
	private final CoveredCallType type;

	public CoveredCall(Option option, CoveredCallType type) {
		this.option = option;
		this.type = type;
	}

	public Option getOption() {
		return this.option;
	}

	public CoveredCallType getType() {
		return this.type;
	}
}
